docs/docsite/rst/dev_guide/testing.rst

```diff
@@ -0,0 +1,200 @@
+
+If you haven't already got Ansible available use the local checkout by doing::
+
+  source hacking/env-setup
+
+Then run the tests detailed in the GitHub comment::
+
+  ansible-test sanity --test pep8
+  ansible-test sanity --test validate-modules
+
+
+If there isn't a GitHub comment stating what's failed you can inspect the results by clicking on the "Details" button under the "checks have failed" message at the end of the PR.
+
+Rerunning a failing CI job
+--------------------------
+
+Occasionally you may find your PR fails due to a reason unrelated to your change, this could happen due to:
+
+* A temporary issue accessing an external resource, such as a yum or git repo
+* A timeout creating a virtual machine to run the tests on
+
+If this appears to be the case you can rerun the Shippable test by:
+
+* Closing and re-opening the PR
+* Making another change to the PR and pushing to GitHub
+
+If the issue continues please contact us in ``#ansible-devel`` on Freenode IRC.
+
+
+How to test a PR
+================
+
+If you're a developer, one of the most valuable things you can do is look at the GitHub issues list and help fix bugs.  We almost always prioritize bug fixing over
+feature development, so clearing bugs out of the way is one of the best things you can do.
+
+Even if you're not a developer, helping test pull requests for bug fixes and features is still immensely valuable.
+
+This goes for testing new features as well as testing bug fixes.
+
+In many cases, code should add tests that prove it works but that's not ALWAYS possible and tests are not always comprehensive, especially when a user doesn't have access
+to a wide variety of platforms, or that is using an API or web service.
+
+In these cases, live testing against real equipment can be more valuable than automation that runs against simulated interfaces.
+In any case, things should always be tested manually the first time too.
+
+Thankfully helping test ansible is pretty straightforward, assuming you are already used to how ansible works.
+
+Setup: Checking out a Pull Request
+----------------------------------
+
+You can do this by checking out ansible, making a test branch off the main one, merging a GitHub issue, testing,
+and then commenting on that particular issue on GitHub. Here's how:
+
+.. warning::
+   Testing source code from GitHub pull requests sent to us does have some inherent risk, as the source code
+   sent may have mistakes or malicious code that could have a negative impact on your system. We recommend
+   doing all testing on a virtual machine, whether a cloud instance, or locally.  Some users like Vagrant
+   or Docker for this, but they are optional.  It is also useful to have virtual machines of different Linux or
```

Comment:
Two spaces before "It". Most sentences seem to have one space (which is preferable, IMO).